src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -2026,15 +2021,164 @@ private bool EvaluateWhenOrOtherwiseChildren(IEnumerable<ProjectElement> childre
+            //      <Import Project='$(ProjectPathForImport)' />
+            //
+
+            // Making a local copy for this instance of Evaluator, to add the the value of
+            // $(MSBuildExtensionsPath*) property that we have here, instead of keeping it
+            // in the static/shared MSBuildExtensionsPathHelper
+            var allExtensionPaths = new List<string>(fallbackExtensionPaths);
+            allExtensionPaths.Insert(0, _data.GetProperty(MSBuildExtensionsPathHelper.ReferenceKindToString(refKindInProject)).EvaluatedValue);
+
+            string extensionPropertyRefAsString = String.Format("$({0})", MSBuildExtensionsPathHelper.ReferenceKindToString(refKindInProject));
+
+            _loggingService.LogComment(_buildEventContext, MessageImportance.Low, "SearchPathsForMSBuildExtensionsPath",
+                                        extensionPropertyRefAsString,
+                                        String.Join(Path.PathSeparator.ToString(), allExtensionPaths));
+
+            bool atleastOneExactFilePathWasLookedAtAndNotFound = false;
+
+            // Try every extension search path, till we get a Hit:
+            // 1. 1 or more project files loaded
+            // 2. 1 or more project files *found* but ignored (like circular, self imports)
+            foreach(var extensionPath in allExtensionPaths) {
+                var newExpandedCondition = importElement.Condition.Replace(extensionPropertyRefAsString, extensionPath);
+                if(!EvaluateConditionCollectingConditionedProperties(importElement, newExpandedCondition, ExpanderOptions.ExpandProperties, ParserOptions.AllowProperties,
+                            _projectRootElementCache))
+                    continue;
+
+                var newExpandedImportPath = importElement.Project;
+                newExpandedImportPath = newExpandedImportPath.Replace(extensionPropertyRefAsString, extensionPath);
+
+                _loggingService.LogComment(_buildEventContext, MessageImportance.Low, "TryingExtensionsPath", newExpandedImportPath, extensionPath);
+
+                bool atleastOneImportWasIgnored = false;
+                bool atleastOneImportProcessed = false;
+                var projects = ExpandAndLoadImportsFromUnescapedImportExpression(directoryOfImportingFile, importElement, newExpandedImportPath, false,
+                                    out atleastOneImportWasIgnored, out atleastOneImportProcessed);
+
+                if(atleastOneImportWasIgnored) {
+                    // Circular, Self import cases are usually ignored
+                    // Since we have a semi-success here, we stop looking at
+                    // other paths
+                    return new List<ProjectRootElement>();
+                }
+
+                // Nothing was ignored, but if @atleastOneImportProcessed==true, then we did try
+                // to look at a file path and found it missing!
```

Comment:
Why the conclusion that it was missing? atleastOneImportProcessed is set to true right at the beginning of the loop, and it may also mean that that path was found and processed into a ProjectRootElement